.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,33 @@
+name: Execute the project's unit tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10']
+
```

Comment:
Why are you only testing one version of Python. You could use the matrix in this Github Action to test different versions of the language.
In addition to justifying the versions chosen.

Reply:
This is explained in the documentation of the project, in the `cicd_notes.md` document. The project is compatible with Python 3.9 and 3.10, and since Python 3.9 was tested using AppVeyor I only tested version 3.10 with the GitHub Action.

Reply:
A matrix with a single value does not make a lot of sense. If you test only one, just use that one.